### ISSUE TITLE
Resize event on stdout not invoked in TTY on macOS

### ISSUE THREAD

**Issue by samholmes** [denoland/deno#26671]
Version: Deno 2.0.4
macOS: 14.2.1

# Repro steps

1. `deno` in Alacritty
2. `process.stdout.on('resize', () => { console.log(`${process.stdout.columns}x${process.stdout.rows}`) })`
3. Resize Alacritty window

# Expected

Printed console output with the process columns and rows.

# Actual

Nothing happens.

# Additionally

If you run the same `process.stdout.on('resize', () => { console.log(`${process.stdout.columns}x${process.stdout.rows}`) })` in `node` (I'm using v20), then you'll see the expected results.

Also, I've tested in Warp and Terminal TTY apps as well and got the same bug.

**Comment by littledivy** [comment]
Update:

PR #27363 hooks up resize signal listeners, this works with `deno run` and friends. 

It didn't work on REPL because rustyline would swallow SIGWINCH signals. Switching to the rustyline `signal_hook` features fixes the issue in the REPL. 

Unfortunately, this causes PTY-based repl tests to timeout because reads would never appear. https://github.com/denoland/deno/actions/runs/14018704803/job/39247719404?pr=27363

Ref https://github.com/kkawakam/rustyline/issues/690
Ref https://github.com/kkawakam/rustyline/issues/703

**Comment by OIRNOIR** [comment]
This seems like it's still broken, and not just on macOS. I'm experiencing this issue in Konsole on KDE Plasma. Any progress on this issue?

**Comment by ddmoney420** [comment]
I have a fix for this issue! 

The problem was that  in the Node.js TTY polyfill wasn't listening for the SIGWINCH signal, which is emitted when the terminal is resized.

## Solution
I've implemented SIGWINCH signal handling in  that:
1. Listens for SIGWINCH signals on Unix platforms (macOS, Linux, etc.)
2. Updates the `columns` and `rows` properties when terminal is resized
3. Emits the 'resize' event to notify listeners

The fix gracefully handles platforms without SIGWINCH support like Windows.

## Testing
Added a test case that verifies the resize event is emitted when SIGWINCH is received.

Branch: https://github.com/ddmoney420/deno/tree/fix-tty-resize-event

Let me know if you'd like me to open a PR or make any adjustments!

**Comment by OIRNOIR** [comment]
The above comment and code smells AI-generated to me. I'm not sure what Deno's policy on AI-generated code is, but it's something to be aware of. In any case, the contribution should be tested to ensure that it actually fixes the issue at hand.

Plus, I don't think we need a new PR for this. We need to revive this old one: https://github.com/denoland/deno/pull/27363

**Comment by ddmoney420** [comment]
You're right on both counts - I did use AI assistance, and I should have checked for existing PRs first. Apologies for the noise.

Deleting my branch and deferring to #27363. I really am sorry for stepping on toes, I appreciate your input!

**Comment by bartlomieju** [comment]
@ddmoney420 if your fix actually works then you are more than welcome to open a separate PR for it

**Comment by ddmoney420** [comment]
Thanks @bartlomieju\! I'd be happy to open a PR for this. I'll wait until my current PR #31985 is merged or closed first to avoid flooding the repo. Will follow up then.
